### PR TITLE
changing jump detection step to use detect_jumps in stcal

### PR DESCRIPTION
**Pull request by cshanahan1**
The main `detect_jumps` function and the two point difference algorithm are now in stcal, this PR makes the necessary changes to use that code.

### CODE REVIEW

**Comment by dmggh** [comment]
Please hold off merging this, as I am in the process of moving the common jump code to stcal via RCAL-152 ("Move the common jump code to STCAL"), which I plan to merge next week.

**Comment by dmggh** [comment]
Again, please hold off merging this, as I am in the process of moving the common jump code to stcal via RCAL-152 ("Move the common jump code to STCAL"), which I plan to merge in the next few days. Thanks.

**Comment by nden** [comment]
- In `setup.cfg` change stcal to point to github/main, e.g.
   `stcal@git+https://github.com/spacetelescope/stcal.git`
We want to demonstrate tests pass with the master branch of stcal before making an stcal release.
- Needs a change log, and the usual links to issues (see PR template)

**Comment by hbushouse** [comment]
Is this ready for review? Or does it need updates due to the on-going work of moving the jump step to stcal?

**Comment by cshanahan1** [comment]
@hbushouse I made a few updates, i'm going to push those changes soon then it will be ready for review

File: jwst/jump/jump.py
Comment (jdavies-st) [inline jwst/jump/jump.py:42]: Does this mean `jump` multiprocessing doesn't work right now?  Is there an existing test for this that's commented out?  Or no test?

Comment (cshanahan1) [reply jwst/jump/jump.py:42]: Multiprocessing is not enabled right now. There are some tests - `test_flagging_of_CRs_across_slice_boundaries` for example. Should I comment them out?

File: setup.cfg
Comment (cshanahan1) [reply setup.cfg:75]: oh whoops, i think thats from trying to resolve a conflict in that file, ill fix it.

File: CHANGES.rst
Comment (hbushouse) [inline CHANGES.rst:36]: this line is a (munged) duplicate of line 19 above and should be removed (it applies to cube_build, not extract_1d). Probably got messed up during conflict resolution.

File: jwst/jump/jump.py
Comment (hbushouse) [inline jwst/jump/jump.py:1]: minor nit, but can get rid of the empty line here

File: jwst/jump/jump.py
Comment (hbushouse) [reply jwst/jump/jump.py:42]: Yes, if MP is not possible to run, then the tests that use it need to be skipped. Can use the "@pytest.mark.skip()" directive, rather than commenting them out.

**Comment by nden** [comment]
@cshanahan1 Please rebase to pick up the rtd fix.

**Review by nden — APPROVED** [review]
lgtm

**Review by hbushouse — APPROVED** [review]

**Comment by hbushouse** [comment]
Since merging this PR the regtest system under Jenkins is failing to build the jwst package with:
Failed to build pymssql-linux
ERROR: Could not build wheels for pymssql-linux which use PEP 517 and cannot be installed directly

Does this have to do with the change in setup.cfg that switched from pymssql to pymssql-linux?

**Comment by jdavies-st** [comment]
Yes, it does.  It was caught in my review above

https://github.com/spacetelescope/jwst/pull/6089#discussion_r673364112

but not fixed before being merged.

**Comment by cshanahan1** [comment]
I'm trying to figure out if I forgot to push my branch after addressing some small final review comments, sorry about that. I'll open a new PR if that's the case.

**Pull request merged**
